bot/exts/moderation/infraction/_scheduler.py

```diff
@@ -76,7 +87,46 @@ async def cog_load(self) -> None:
 
             self.scheduler.schedule_at(next_reschedule_point, -1, self.cog_load())
 
-        log.trace("Done rescheduling")
+        log.trace("Done rescheduling expirations, scheduling tidy up tasks.")
+
+        for key, expire_at in await self.messages_to_tidy.items():
+            channel_id, message_id = map(int, key.split(":"))
+            expire_at = dateutil.parser.isoparse(expire_at)
+
+            log.trace(
+                "Scheduling tidy up for message %s in channel %s at %s",
+                message_id, channel_id, expire_at
+            )
+
+            self.tidy_up_scheduler.schedule_at(
+                expire_at,
+                message_id,
+                self._delete_infraction_message(channel_id, message_id)
+            )
+
+    async def _delete_infraction_message(
+        self,
+        channel_id: int,
+        message_id: int
+    ) -> None:
+        """
+        Delete a message in the given channel.
+
+        This is used to delete infraction messages after a certain period of time.
+        """
+        channel = await get_or_fetch_channel(self.bot, channel_id)
+        if channel is None:
+            log.warning(f"Channel {channel_id} not found for infraction message deletion.")
+            return
+
+        try:
+            message = await channel.fetch_message(message_id)
+            await message.delete()
```

Comment:
Can we avoid the message fetch call? Can we delete the message if we just know its ID (and channel/guild IDs)?

Reply:
Without hooking into the internals of discord.py, there isn't a way without fetching the message (you could in theory either call the raw HTTP message or try coerce a `discord.Object` into being a `discord.Message`). I couldn't find a particularly clean way though so settled for just fetching the message.

Reply:
No worries, it's not a major concern. We can revisit this if it proves to be a performance problem.